CHANGELOG.md

```diff
@@ -2,6 +2,165 @@
+
+#### Remove usage of the `GAS` opcode in module execute flows
+
+Issue: [#459](https://github.com/safe-global/safe-contracts/issues/459)
+
+The following rule of usage of the `GAS` opcode in the ERC-4337 standard made it impossible to build a module to support ERC4337:
+
+> -   Must not use GAS opcode (unless followed immediately by one of { CALL, DELEGATECALL, CALLCODE, STATICCALL }.)
+
+We removed the `GAS` opcode usage in module transactions to forward all the available gas instead.
+
+#### Require the `to` address to be a contract in `setupModules`
+
+Issue: [#483](https://github.com/safe-global/safe-contracts/issues/483)
+
+The `setupModules` method was changed to require the `to` address to be a contract. If the `to` address is not a contract, the transaction will be reverted with a `GS002` error code.
+
+#### Enforce the `dataHash` is equal to `data` in the signature verification process for contract signatures
+
+Issue: [#497](https://github.com/safe-global/safe-contracts/issues/497)
+
+To prevent unexpected behaviour, the `dataHash` must now equal a hash of the `data` in the signature verification process for contract signatures. Otherwise, the transaction will be reverted with a `GS027` error code.
+
+#### Fix `getModulesPaginated` to return a correct `next` pointer
+
+Issue: [#461](https://github.com/safe-global/safe-contracts/issues/461)
+
+The `getModulesPaginated` method was fixed to return a correct `next` pointer. The `next` pointer now equals the last module in the returned array.
+
+#### Check the EIP-165 signature of the Guard before adding
+
+Issue: [#309](https://github.com/safe-global/safe-contracts/issues/309)
+
+When setting a guard, the core contract will check that the target address supports the Guard interface with an EIP-165 check. If it doesn't, the transaction will revert with the `GS300` error code.
+
+#### Index essential parameters when emitting events
+
+Issue: [#541](https://github.com/safe-global/safe-contracts/issues/541)
+
+Index essential parameters in the essential events, such as:
+
+-   Owner additions and removals (Indexed parameter - owner address)
+-   Fallback manager changes (Indexed parameter - fallback manager address)
+-   Module additions and removals (Indexed parameter - module address)
+-   Transaction guard changes (Indexed parameter - guard address)
+-   Transaction execution/failure (Indexed parameter - transaction hash)
+
+### Factory
+
+Umbrella issue: [#462](https://github.com/safe-global/safe-contracts/issues/462)
+
+#### Remove the `createProxy` method
+
+This method uses the `CREATE` opcode, which is not counterfactual for a specific deployment. This caused user errors and lost/stuck funds and is now removed.
+
+#### Add a check that singleton exists for the initializer call
+
+If the initializer data is provided, the factory now checks that the singleton contract exists and the success of the call to avoid a proxy being deployed uninitialized
```

Comment:
```suggestion
If the initializer data is provided, the Factory now checks that the Singleton contract exists and the success of the call to avoid a proxy being deployed uninitialized
```